x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/configuration_form/source_field.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiLink, EuiSpacer, EuiCallOut } from '@elastic/eui';
+
+import { documentationService } from '../../../../services/documentation';
+import {
+  UseField,
+  FormDataProvider,
+  FormRow,
+  ToggleField,
+  ComboBoxField,
+  FieldHook,
+} from '../../shared_imports';
+import { ComboBoxOption } from '../../types';
+
+const i18nTexts = {
+  sourceField: {
+    title: i18n.translate('xpack.idxMgmt.mappingsEditor.sourceFieldTitle', {
+      defaultMessage: '_source field',
+    }),
+    description: (
+      <FormattedMessage
+        id="xpack.idxMgmt.mappingsEditor.sourceFieldDescription"
+        defaultMessage="The _source field contains the original JSON document body that was provided at index time. Individual fields can be pruned by defining which ones to include or exclude from the _source field. {docsLink}"
+        values={{
+          docsLink: (
+            <EuiLink href={documentationService.getMappingSourceFieldLink()} target="_blank">
+              {i18n.translate('xpack.idxMgmt.mappingsEditor.sourceFieldDocumentionLink', {
+                defaultMessage: 'Learn more.',
+              })}
+            </EuiLink>
+          ),
+        }}
+      />
+    ),
+  },
+  comboBox: {
+    placeholder: i18n.translate(
+      'xpack.idxMgmt.mappingsEditor.sourceIncludeField.placeholderLabel',
+      {
+        defaultMessage: 'path.to.field.*',
+      }
+    ),
+  },
+};
+
+const ComboBoxFields = ({ field }: { field: FieldHook }) => {
+  const [options, setOptions] = useState<ComboBoxOption[]>([]);
+  const { setValue } = field;
+
+  return (
+    <ComboBoxField
```

Comment:
The `ComboBoxField` takes care of almost everything. We just need to override below what is unique in this context.